src/main/java/beam/utils/gtfs/ConvertScheduledTripsToFrequency.java

```diff
@@ -0,0 +1,81 @@
+package beam.utils.gtfs;
+
+import beam.router.r5.NetworkCoordinator;
+import beam.sim.BeamHelper;
+import beam.utils.BeamConfigUtils;
+import com.conveyal.r5.analyst.scenario.AddTrips;
+import com.conveyal.r5.analyst.scenario.AdjustFrequency;
+import com.conveyal.r5.analyst.scenario.Modification;
+import com.conveyal.r5.analyst.scenario.Scenario;
+import com.conveyal.r5.transit.TransportNetwork;
+import com.typesafe.scalalogging.LazyLogging;
+import com.typesafe.scalalogging.Logger;
+import org.matsim.core.scenario.MutableScenario;
+import scala.Tuple3;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Convert a line to frequency.
+ */
+public class ConvertScheduledTripsToFrequency extends Modification {
+    public static final String type = "convert-to-frequency";
+    @Override
+    public String getType() {
+        return "convert-to-frequency";
+    }
+
+    @Override
+    public boolean apply(TransportNetwork network) {
+        return toR5().apply(network);
+    }
+
+    @Override
+    public int getSortOrder() {
+        return 49;
+    }
+
+    public String feed;
+    public String[] routes;
+
+    /** Should trips on this route that start outside the days/times specified by frequency entries be retained? */
+    public boolean retainTripsOutsideFrequencyEntries = false;
+
+    public List<FrequencyEntry> entries;
+
+    public static class FrequencyEntry extends AbstractTimetable {
+        /** start times of this trip (seconds since midnight), when non-null scheduled trips will be created */
+        @Deprecated
```

Comment:
Why create a new variable that is automatically deprecated? Can we just remove this?

Reply:
In this PR I am just synchronizing two branches. I am not changing the code from `competitions-pin_dev`